tests/FeatureModelCastTest.php

```diff
@@ -96,6 +96,39 @@ public function it_saves_a_null_value_for_nullable_array_of_enums()
         $this->assertNull($model->nullable_array_of_enums);
     }
 
+    /** @test */
+    public function it_saves_a_null_value_for_nullable_set_of_enums()
+    {
+        $model = Post::create([
+            'status' => StatusEnum::draft(),
+            'nullable_set_of_enums' => null,
+        ]);
+
+        $model->refresh();
+
+        $this->assertTrue($model->status->equals(StatusEnum::draft()));
+        $this->assertEquals('draft', $model->getOriginal('status'));
+        $this->assertNull($model->nullable_set_of_enums);
+    }
+
+    /** @test */
+    public function it_saves_an_enum_of_set_of_enums()
+    {
+        $model = Post::create([
+            'status' => StatusEnum::draft(),
+            'set_of_enums' => [StatusEnum::draft(), StatusEnum::archived()],
+        ]);
+
+        $model->refresh();
+
+        $this->assertTrue($model->status->equals(StatusEnum::draft()));
+        $this->assertEquals('draft', $model->getOriginal('status'));
+        $this->assertIsArray($model->set_of_enums);
+        $this->assertCount(2, $model->set_of_enums);
```

Comment:
An assertion to the raw attribute is missing. Right now this test would also pass if the JSON formater is applied. So it should check that the raw attribute is `DRAFT,ARCHIVED` and in the best case add a similar assertion to one of the JSON ones.

Reply:
good point will add the assertion shortly

Reply:
it's done and waiting for your review

Reply:
Hey, @Gummibeer hope you're doing well! Have you been able to check the PR?